packages/blocks/src/__internal__/utils/common-operations.ts

```diff
@@ -6,10 +6,14 @@ import type { Quill } from 'quill';
 import type { ExtendedModel } from './types.js';
 
 // XXX: workaround quill lifecycle issue
-export function asyncFocusRichText(page: Page, id: string) {
+export function asyncFocusRichText(page: Page, id: string, retries = 2) {
   requestAnimationFrame(() => {
     const adapter = page.richTextAdapters.get(id);
-    adapter?.quill.focus();
+    if (adapter) {
+      adapter?.quill.focus();
+    } else if (retries > 0) {
+      asyncFocusRichText(page, id, retries - 1);
+    }
```

Comment:
It looks like the DOM is mounted first, then the quill is initialized.

Can we focus on the DOM directly instead of relying on quill API?

Reply:
hmm, I think `focus` here will also mean to place the cursor onto the contenteditable right? When quill is not being appended, the DOM is just an empty placeholder without any contents.